webhooks/clusterclass.go

```diff
@@ -99,6 +103,19 @@ func (webhook *ClusterClass) ValidateUpdate(_ context.Context, oldObj, newObj ru
 
 // ValidateDelete implements validation for ClusterClass delete.
 func (webhook *ClusterClass) ValidateDelete(ctx context.Context, obj runtime.Object) error {
+	clusterClass, ok := obj.(*clusterv1.ClusterClass)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a ClusterClass but got a %T", obj))
+	}
+	clusters, err := webhook.clustersUsingClusterClass(clusterClass)
+	if err != nil {
+		return err.ToAggregate()
```

Comment:
Should we use `apierrors.NewInvalid(clusterv1.GroupVersion.WithKind("ClusterClass").GroupKind(), new.Name, allErrs)` like in the other webhooks?

Reply:
+1 let's stick with apierrors as a containers for our errors